### PR TITLE
Fixes some material datum bugs

### LINKED ISSUE #47004: Putting floor tiles in a protolathe gives 2000 of that material instead of 500.
## Reproduction:

Just take a sheet out of the protolathe, turn it into 4 floor tiles, put it back in the protolathe and watch as 3 extra sheets of material appear.

https://i.imgur.com/qmIU1W4.png

floydmats was well tested

### LINKED ISSUE #47005: Explosion logging is way too verbose
## Round ID:
Round ID: 121074
BYOND Version: 512.1484
Server revision compiled on: 
Master commit: e49ebbbd8788896cbc9b8d8b79a440e471fe53e8
Local commit: b11d0a6c130b451b7ae9db1f9d6471b55cb27d12
Server tools version: /tg/station 13 Server v3.2.6.2

## Reproduction:

Explosion logs are acting really weird and I think are creating new lines for each tile affected by an explosion (note the somewhat iterative numbering on the coords mentioned)

Step 1: be admin
Step 2: thing explodes
Step 3: logs!
[](https://i.imgur.com/rS1Y4Eu.png)

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
 fix: You now get the correct material from deconning tables and chairs
 fix: Objects now get the correct amount of materials
 fix: Plasma mat doesn't send warnings on equip
 fix: Plasma alloys dont explode anymore
 fix: no more mat duping for now hopefully
/:cl:

Passing in a list of materials to set_custom_materials() that weren't instanced would make it try to find the amount by inputting the singleton, because the list uses types, it will always return null/0 because theres nothing in that slot. Instead it now finds the amount by inputing the iterator (which can be either instanced or non instanced, but always fits the list)

closes #47004
closes #47005

### CODE REVIEW

**Comment by Couls** [comment]
Make sure this fixes deconstructing chairs too I remember getting metal instead of bananium from bananium chairs

**Comment by Qustinnus** [comment]
done

**Comment by PKPenguin321** [comment]
https://github.com/tgstation/tgstation/issues/47005 this one seems kind of important if you're up for another. seems to be caused by plasteel having a tiny amount of plasma

**Comment by Qustinnus** [comment]
theoretically fixed if @Fikou is up to make alloys into proper datums otherwise i can do a quickfix tommorow

**Comment by FancyCyclops** [comment]
Have you tried testing your code before getting it merged?

**Comment by Qustinnus** [comment]
I realize you've never programmed before in your life but literaly noone will come up with the test case "I made everything out of materials, I should see what happens when I put tiles into the protolathe".

**Comment by Qustinnus** [comment]
travis had a stroke, can the build on this be restarted?

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 1]
```diff
@@ -56,6 +56,10 @@
 	// If we have materials, and don't have the NOCONSTRUCT flag
 	if(buildstacktype && (!(flags_1 & NODECONSTRUCT_1)))
 		new buildstacktype(loc,buildstackamount)
+	else
+		for(var/i in custom_materials)
```
Comment (AnturK) [inline code/game/objects/structures/beds_chairs/chair.dm]: Should this be ignoring NODECONSTRUCT ?

**Review by AnturK — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**